app/ts-cli/geminicli/cli.go

```diff
@@ -0,0 +1,375 @@
+/*
+Copyright 2022 Huawei Cloud Computing Technologies Co., Ltd.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package geminicli
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"os"
+	"os/signal"
+	"sort"
+	"strings"
+	"syscall"
+	"time"
+
+	"github.com/c-bata/go-prompt"
+	"github.com/influxdata/influxdb/client"
+	"github.com/influxdata/influxdb/models"
+	"github.com/jedib0t/go-pretty/v6/table"
+	"github.com/openGemini/openGemini/app/ts-cli/geminiql"
+	"github.com/openGemini/openGemini/open_src/influx/influxql"
+)
+
+const (
+	CLIENT_VERSION = "0.1.0"
+)
+
+var (
+	FilePathCompletionSeparator = string([]byte{' ', os.PathSeparator})
+)
+
+type CommandLineConfig struct {
+	Host             string
+	Port             int
+	UnixSocket       string
+	Username         string
+	Password         string
+	Database         string
+	Type             string
+	Ssl              bool
+	IgnoreSsl        bool
+	Format           string
+	Precision        string
+	WriteConsistency string
+	Pretty           bool
+
+	PathPrefix string
+}
+
+type CommandLineFactory struct {
+}
+
+func (f CommandLineFactory) CreateCommandLine(config CommandLineConfig) (*CommandLine, error) {
+	c := &CommandLine{
+		cliConfig: config,
+		osSignals: make(chan os.Signal, 1),
+		parser:    geminiql.QLNewParser(),
+	}
+
+	addr := fmt.Sprintf("%s:%d/%s", config.Host, config.Port, config.PathPrefix)
+	url, err := client.ParseConnectionString(addr, config.Ssl)
+	if err != nil {
+		return nil, err
+	}
+	c.url = url
+	c.ssl = config.Ssl
+
+	c.config.UnixSocket = config.UnixSocket
+	c.config.Username = config.Username
+	c.config.Password = config.Password
+	c.config.UnsafeSsl = config.IgnoreSsl
+	c.config.Precision = config.Precision
+	c.config.WriteConsistency = config.WriteConsistency
```

Comment:
Looks like the comand line options are compatible with influx, please mention it
in comments, and make sure the semantics are the same.